plugins/org.locationtech.udig.tool.edit/src/org/locationtech/udig/tools/edit/behaviour/accept/AcceptChangesBehaviour.java

```diff
@@ -263,20 +265,23 @@ public UndoableMapCommand getCommand(EditToolHandler handler) {
                     // commands.add(newFeatureCommand);
                     // }
                 } else {
-                    // not creating it so don't need to set it.
-                    UndoableMapCommand setGeometryCommand = new SetGeometryCommand(editGeom
-                            .getFeatureIDRef().get(), new StaticBlockingProvider<ILayer>(layer),
-                            SetGeometryCommand.DEFAULT, geom);
-                    commands.add(setGeometryCommand);
+                    String idRef = editGeom.getFeatureIDRef().get();
+                    if (idRef != null) {
```

Comment:
@sschulz92 I'm wondering about why the check is not implemented everywhere `editGeom.getFeatureIDRef().get()` occurs within this class.

I would expect to check in the other else-block down below as well. Therfore I think its worth to assign it to a new variable once and use it in both blocks to verify of none null.

Reply:
Can you have a second look please? I just pushed the change in order to fulfill your review.